src/pipx/interpreter.py

```diff
@@ -58,6 +61,14 @@ def find_python_interpreter(python_version: str) -> str:
 
     if shutil.which(python_version):
         return python_version
+
+    if fetch_missing_python or FETCH_MISSING_PYTHON:
+        try:
+            standalone_executable = download_python_build_standalone(python_version)
+            return standalone_executable
+        except PipxError as e:
+            raise InterpreterResolutionError(source="the python-build-standalone project", version=python_version) from e
```

Comment:
```suggestion
        except PipxError as e:
            logger.error(e)
            raise InterpreterResolutionError(source="the python-build-standalone project", version=python_version) from e
```

Should probably log the original error here. You've created nice errors with a lot of detail that would otherwise just not be shown, as the InterpreterResolutionError is caught and just it's content is printed.

Reply:
The benefit of raising an exception *from* another exception (ie. `raise InterpreterResolutionError(...) from e`) is that the stack trace includes the surface error *and* the underlying error.

I think logging the error on top of that would be excessive

Reply:
but the PipxError is caught in `pipx/main`, line 191, so the stack trace will never be printed?

```
(develop-3-12) ➜  pipx git:(resolve_home_path_in_dir_env_vars) ✗ pipx install --python python99.99 -vv
pipx >(setup:858): 2024-02-06 16:43:45
pipx >(setup:859): /home/robert/private/code/pipx/.nox/develop-3-12/bin/pipx install --python python99.99 -vv
pipx >(setup:860): pipx version is 1.4.4.dev16+g3e423c2.d20240203
pipx >(setup:861): Default python interpreter is '/home/robert/private/code/pipx/.nox/develop-3-12/bin/python'
⚠️ No executable for the provided Python version 'python99.99' found in PATH. Please make sure the provided version is on your PATH or the file path is valid.
pipx >(cli:932): pipx finished.
```

Reply:
ahhhhh, i see.

if `run_pipx_command` is swallowing exception details, wouldn't it be better to have `run_pipx_command` log those exception details rather than to add special handling to all functions which raise exception chains?

ie. add `log.error(e, exc_info=True)` to the exception-handling code in `run_pipx_command`

Reply:
Fair enough. I'd set it to `logger.info` though, as we would only want to log the traceback if the command is executed with verbose. The normal print should be enough information for normal use.

Reply:
do you want me to do this as part of the PR?

I would prefer it be done by one of the maintainers as a separate thing... it feels like something that belongs outside the scope of this PR, but if you disagree, i'm willing to write and commit the change

Reply:
Sure, I can do it in a separate PR. I don't see any harm in doing it here, but it's up to you.